### PR TITLE
Java random forest

### PR DESCRIPTION
**Pull request by alexandraj777**

### CODE REVIEW

File: random-forest/src/main/java/com/example/RandomForestApp.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        .type("int")
+                        .bounds(new Bounds(1.0, 100.0))
+                        .build()
+                ))
+                .observationBudget(60)
+                .build())
+            .call();
+        System.out.println("Created experiment: https://sigopt.com/experiment/" + experiment.getId());
+
+        // Run the Optimization Loop between 10x - 20x the number of parameters
```
Comment (pfhayes) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: would this comment be more appropriate where observation budget is declared?

File: random-forest/src/main/java/com/example/RandomForestApp.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        // Re-fetch the experiment to get the best observed value and assignments
+        experiment = Experiment.fetch(experiment.getId()).call();
+        Assignments bestAssignments = experiment.getProgress().getBestObservation().getAssignments();
+        RandomForest bestRandomForest = new RandomForest();
+
+        // To wrap up the Experiment, fit the RandomForest on the best assigments and train on all available data
+        bestRandomForest.setMaxDepth(((Double)bestAssignments.get("maxDepth")).intValue());
+        bestRandomForest.setBagSizePercent(((Double)bestAssignments.get("bagSizePercent")).intValue());
+        bestRandomForest.setNumIterations(((Double)bestAssignments.get("numIterations")).intValue());
+        bestRandomForest.buildClassifier(irisData);
```
Comment (pfhayes) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: should this just be a call to evaluateModeL?

Comment (pfhayes) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: or maybe just abstract out the contruction of a forest from the assignments?

File: random-forest/src/main/java/com/example/RandomForestApp.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        // Re-fetch the experiment to get the best observed value and assignments
+        experiment = Experiment.fetch(experiment.getId()).call();
+        Assignments bestAssignments = experiment.getProgress().getBestObservation().getAssignments();
+        RandomForest bestRandomForest = new RandomForest();
+
+        // To wrap up the Experiment, fit the RandomForest on the best assigments and train on all available data
+        bestRandomForest.setMaxDepth(((Double)bestAssignments.get("maxDepth")).intValue());
+        bestRandomForest.setBagSizePercent(((Double)bestAssignments.get("bagSizePercent")).intValue());
+        bestRandomForest.setNumIterations(((Double)bestAssignments.get("numIterations")).intValue());
```
Comment (pfhayes) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: maybe we want to add a... .getInt or something to Assignments

File: random-forest/src/main/java/com/example/RandomForestApp.java (hunk 2 above)
Comment (alexandraj777) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: It's a little different, here we are training the model with all of the available data, whereas in evaluateModel we're holding out data on every cross validation fold and testing.

File: random-forest/src/main/java/com/example/RandomForestApp.java (hunk 1 above)
Comment (alexandraj777) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: Sure we can do that! I got into the habit of including it at the beginning of the optimization loop

File: random-forest/src/main/java/com/example/RandomForestApp.java (hunk 3 above)
Comment (alexandraj777) [inline random-forest/src/main/java/com/example/RandomForestApp.java]: Yes, maybe `getInt` `getDouble` `getCategorical`?

**Pull request merged**